Rustdoc: include crate name in links for local primitives

gurry opened the pull request:
Fixes #121106. 

This change makes links to primitives easier to use when the path of the page where they will be embedded is not known beforehand such as when we generate impls dynamically from the `register_type_impls` method in `main.js`, which is exactly what is happening in #121106. 

An example to show the effect of this change: earlier, if the current page in `cx.current` inside `primitive_link_fragment()` was `std::simd::prelude::Simd` the generated path would be `../../primitive.<prim>.html`. Now it would be `../../../std/primitive.<prim>.html` instead.

A side effect of the change is that local primitive links _everywhere_ will now contain the crate name, even outside of the dynamic situation mentioned above. I'm not sure if there are any major downsides of that other than making the links a bit longer. Ideally I wanted to restrict this behaviour change to only the dynamic cases. We could have achieved that by passing an additional bool arg to `primitive_link_fragment()`, but it felt awkward to do so. Any alternative suggestions are welcome.

gurry commented:
cc @notriddle

gurry commented:
> > Ideally I wanted to restrict this behaviour change to only the dynamic cases.
> 
> To avoid making the regular standard library docs bigger, could this be done based on when `cx.current.len() == 0`?
> 
> ```rust
> match m.primitive_locations.get(&prim) {
>     Some(&def_id) if def_id.is_local() && cx.current.is_empty() => {
>         write!(
>             f,
>             "<a class=\"primitive\" href=\"{}/primitive.{}.html{fragment}\">",
>             ExternalCrate { crate_num: def_id.krate }.name(cx.tcx()),
>             prim.as_sym()
>         )?;
>         needs_termination = true;
>     }
>     Some(&def_id) if def_id.is_local() => {
>         write!(
>             f,
>             "<a class=\"primitive\" href=\"{}primitive.{}.html{fragment}\">",
>             "../".repeat(cx.current.len() - 1),
>             prim.as_sym()
>         )?;
>         needs_termination = true;
>     }
> ```

Thanks for the suggestion. I'll make the change.

gurry commented:
> Thanks for the suggestion. I'll make the change.

Done

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8027987965/job/21932632685) [(plain)](https://github.com/rust-lang/rust/commit/4722344cf7a457396318b298d94379f5d07ea77b/checks/21932632685/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=gurry
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_d4509e9d-4575-4db5-a180-c533eeae82f0
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=121106-broken-usize-link-in-docs
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_d4509e9d-4575-4db5-a180-c533eeae82f0
GITHUB_REF=refs/pull/121490/merge
GITHUB_REF_NAME=121490/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=9d47cd3c9471d9734fdea216b7613b1f42937bdb
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_d4509e9d-4575-4db5-a180-c533eeae82f0
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_d4509e9d-4575-4db5-a180-c533eeae82f0
GITHUB_TRIGGERING_ACTOR=gurry
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/121490/merge
GITHUB_WORKFLOW_SHA=9d47cd3c9471d9734fdea216b7613b1f42937bdb
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
#12 writing image sha256:6f445f691f20055a49e6cb63a6cea6cdb1dc0f39af99728a5e792078c675077e done
#12 naming to docker.io/library/rust-ci done
#12 DONE 10.2s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Sat Feb 24 03:42:45 UTC 2024
  network time: Sat, 24 Feb 2024 03:42:45 GMT
  network time: Sat, 24 Feb 2024 03:42:45 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'build.optimized-compiler-builtins', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
##[endgroup]
Testing GCC stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling y v0.1.0 (/checkout/compiler/rustc_codegen_gcc/build_system)
    Finished release [optimized] target(s) in 1.23s
     Running `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-codegen/x86_64-unknown-linux-gnu/release/y test --use-system-gcc --use-backend gcc --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc --release --no-default-features --mini-tests --std-tests`
Using system GCC
Using system GCC
[BUILD] example
[AOT] mini_core_hello_world
/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc/mini_core_hello_world
abc
---
##[group]Building stage0 tool linkchecker (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.11s
##[endgroup]
##[group]Testing stage0 Linkcheck (x86_64-unknown-linux-gnu)
core/any/trait.Any.html:22: broken link - `primitive.bool.html`
core/any/trait.Any.html:36: broken link - `primitive.reference.html`
core/any/trait.Any.html:51: broken link - `primitive.reference.html`
core/any/trait.Any.html:70: broken link - `primitive.reference.html`
core/any/trait.Any.html:84: broken link - `primitive.reference.html`
core/any/trait.Any.html:100: broken link - `primitive.bool.html`
core/any/trait.Any.html:114: broken link - `primitive.reference.html`
core/any/trait.Any.html:128: broken link - `primitive.reference.html`
core/any/trait.Any.html:146: broken link - `primitive.reference.html`
core/any/trait.Any.html:159: broken link - `primitive.reference.html`
core/any/trait.Any.html:174: broken link - `primitive.bool.html`
core/any/trait.Any.html:188: broken link - `primitive.reference.html`
core/any/trait.Any.html:202: broken link - `primitive.reference.html`
core/any/trait.Any.html:220: broken link - `primitive.reference.html`
core/any/trait.Any.html:231: broken link - `primitive.reference.html`
core/any/fn.type_name.html:2: broken link - `primitive.str.html`
core/any/struct.TypeId.html:22: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:22: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:22: broken link - `primitive.slice.html`
core/any/struct.TypeId.html:22: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:26: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:26: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:27: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:27: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:28: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:28: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:28: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:28: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:29: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:29: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:29: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:29: broken link - `primitive.bool.html`
core/any/struct.TypeId.html:32: broken link - `primitive.reference.html`
core/any/struct.TypeId.html:33: broken link - `primitive.reference.html`
core/any/fn.type_name_of_val.html:2: broken link - `primitive.reference.html`
core/any/fn.type_name_of_val.html:2: broken link - `primitive.str.html`
core/hint/fn.unreachable_unchecked.html:2: broken link - `primitive.never.html`
core/hint/fn.assert_unchecked.html:2: broken link - `primitive.bool.html`
core/primitive.i32.html:3: broken link - `primitive.i32.html`
core/primitive.i32.html:15: broken link - `primitive.u32.html`
core/primitive.i32.html:18: broken link - `primitive.str.html`
core/primitive.i32.html:18: broken link - `primitive.u32.html`
core/primitive.i32.html:33: broken link - `primitive.u32.html`
core/primitive.i32.html:40: broken link - `primitive.u32.html`
core/primitive.i32.html:45: broken link - `primitive.u32.html`
core/primitive.i32.html:54: broken link - `primitive.u32.html`
core/primitive.i32.html:61: broken link - `primitive.u32.html`
core/primitive.i32.html:68: broken link - `primitive.u32.html`
core/primitive.i32.html:75: broken link - `primitive.u32.html`
core/primitive.i32.html:85: broken link - `primitive.u32.html`
core/primitive.i32.html:189: broken link - `primitive.u32.html`
core/primitive.i32.html:196: broken link - `primitive.u32.html`
core/primitive.i32.html:235: broken link - `primitive.u32.html`
core/primitive.i32.html:242: broken link - `primitive.u32.html`
core/primitive.i32.html:422: broken link - `primitive.u32.html`
core/primitive.i32.html:429: broken link - `primitive.u32.html`
core/primitive.i32.html:442: broken link - `primitive.u32.html`
core/primitive.i32.html:448: broken link - `primitive.u32.html`
core/primitive.i32.html:455: broken link - `primitive.u32.html`
core/primitive.i32.html:468: broken link - `primitive.u32.html`
core/primitive.i32.html:494: broken link - `primitive.u32.html`
core/primitive.i32.html:501: broken link - `primitive.u32.html`
core/primitive.i32.html:529: broken link - `primitive.u32.html`
core/primitive.i32.html:544: broken link - `primitive.u32.html`
core/primitive.i32.html:587: broken link - `primitive.u32.html`
core/primitive.i32.html:602: broken link - `primitive.u32.html`
core/primitive.i32.html:616: broken link - `primitive.u32.html`
core/primitive.i32.html:687: broken link - `primitive.u32.html`
core/primitive.i32.html:698: broken link - `primitive.u32.html`
core/primitive.i32.html:721: broken link - `primitive.u32.html`
core/primitive.i32.html:729: broken link - `primitive.u32.html`
core/primitive.i32.html:737: broken link - `primitive.bool.html`
core/primitive.i32.html:745: broken link - `primitive.bool.html`
core/primitive.i32.html:745: broken link - `primitive.bool.html`
core/primitive.i32.html:782: broken link - `primitive.u32.html`
core/primitive.i32.html:782: broken link - `primitive.bool.html`
core/primitive.i32.html:792: broken link - `primitive.bool.html`
core/primitive.i32.html:800: broken link - `primitive.bool.html`
core/primitive.i32.html:800: broken link - `primitive.bool.html`
core/primitive.i32.html:838: broken link - `primitive.u32.html`
core/primitive.i32.html:838: broken link - `primitive.bool.html`
core/primitive.i32.html:848: broken link - `primitive.bool.html`
core/primitive.i32.html:856: broken link - `primitive.bool.html`
core/primitive.i32.html:866: broken link - `primitive.bool.html`
core/primitive.i32.html:876: broken link - `primitive.bool.html`
core/primitive.i32.html:886: broken link - `primitive.bool.html`
core/primitive.i32.html:896: broken link - `primitive.bool.html`
core/primitive.i32.html:905: broken link - `primitive.u32.html`
core/primitive.i32.html:905: broken link - `primitive.bool.html`
core/primitive.i32.html:914: broken link - `primitive.u32.html`
core/primitive.i32.html:914: broken link - `primitive.bool.html`
core/primitive.i32.html:923: broken link - `primitive.bool.html`
core/primitive.i32.html:935: broken link - `primitive.u32.html`
core/primitive.i32.html:935: broken link - `primitive.bool.html`
core/primitive.i32.html:943: broken link - `primitive.u32.html`
core/primitive.i32.html:1074: broken link - `primitive.u32.html`
core/primitive.i32.html:1084: broken link - `primitive.u32.html`
core/primitive.i32.html:1089: broken link - `primitive.u32.html`
core/primitive.i32.html:1094: broken link - `primitive.u32.html`
core/primitive.i32.html:1102: broken link - `primitive.u32.html`
core/primitive.i32.html:1106: broken link - `primitive.u32.html`
core/primitive.i32.html:1126: broken link - `primitive.u32.html`
core/primitive.i32.html:1149: broken link - `primitive.bool.html`
core/primitive.i32.html:1156: broken link - `primitive.bool.html`
core/primitive.i32.html:1163: broken link - `primitive.u8.html`
core/primitive.i32.html:1163: broken link - `primitive.array.html`
core/primitive.i32.html:1168: broken link - `primitive.u8.html`
core/primitive.i32.html:1168: broken link - `primitive.array.html`
core/primitive.i32.html:1173: broken link - `primitive.u8.html`
core/primitive.i32.html:1173: broken link - `primitive.array.html`
core/primitive.i32.html:1188: broken link - `primitive.u8.html`
core/primitive.i32.html:1188: broken link - `primitive.array.html`
core/primitive.i32.html:1200: broken link - `primitive.u8.html`
core/primitive.i32.html:1200: broken link - `primitive.array.html`
core/primitive.i32.html:1212: broken link - `primitive.u8.html`
core/primitive.i32.html:1212: broken link - `primitive.array.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.reference.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1237: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1238: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.bool.html`
core/primitive.i32.html:1242: broken link - `primitive.i32.html`
core/primitive.i32.html:1242: broken link - `primitive.bool.html`
core/primitive.i32.html:1247: broken link - `primitive.i16.html`
core/primitive.i32.html:1247: broken link - `primitive.i32.html`
core/primitive.i32.html:1247: broken link - `primitive.i16.html`
core/primitive.i32.html:1248: broken link - `primitive.i32.html`
core/primitive.i32.html:1248: broken link - `primitive.i32.html`
core/primitive.i32.html:1249: broken link - `primitive.i32.html`
core/primitive.i32.html:1249: broken link - `primitive.f64.html`
core/primitive.i32.html:1249: broken link - `primitive.i32.html`
core/primitive.i32.html:1250: broken link - `primitive.i32.html`
core/primitive.i32.html:1250: broken link - `primitive.i128.html`
core/primitive.i32.html:1250: broken link - `primitive.i32.html`
core/primitive.i32.html:1251: broken link - `primitive.i32.html`
core/primitive.i32.html:1251: broken link - `primitive.i64.html`
core/primitive.i32.html:1251: broken link - `primitive.i32.html`
core/primitive.i32.html:1252: broken link - `primitive.i8.html`
core/primitive.i32.html:1252: broken link - `primitive.i32.html`
core/primitive.i32.html:1252: broken link - `primitive.i8.html`
core/primitive.i32.html:1253: broken link - `primitive.u16.html`
core/primitive.i32.html:1253: broken link - `primitive.i32.html`
core/primitive.i32.html:1253: broken link - `primitive.u16.html`
core/primitive.i32.html:1254: broken link - `primitive.u8.html`
core/primitive.i32.html:1254: broken link - `primitive.i32.html`
core/primitive.i32.html:1254: broken link - `primitive.u8.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.str.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.reference.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.reference.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1255: broken link - `primitive.i32.html`
core/primitive.i32.html:1258: broken link - `primitive.i32.html`
core/primitive.i32.html:1258: broken link - `primitive.i32.html`
core/primitive.i32.html:1258: broken link - `primitive.bool.html`
core/primitive.i32.html:1259: broken link - `primitive.i32.html`
core/primitive.i32.html:1259: broken link - `primitive.bool.html`
core/primitive.i32.html:1260: broken link - `primitive.i32.html`
core/primitive.i32.html:1260: broken link - `primitive.i32.html`
core/primitive.i32.html:1260: broken link - `primitive.i32.html`
core/primitive.i32.html:1260: broken link - `primitive.bool.html`
core/primitive.i32.html:1260: broken link - `primitive.i32.html`
core/primitive.i32.html:1260: broken link - `primitive.bool.html`
core/primitive.i32.html:1261: broken link - `primitive.i32.html`
core/primitive.i32.html:1261: broken link - `primitive.bool.html`
core/primitive.i32.html:1262: broken link - `primitive.i32.html`
core/primitive.i32.html:1262: broken link - `primitive.bool.html`
core/primitive.i32.html:1262: broken link - `primitive.i32.html`
core/primitive.i32.html:1262: broken link - `primitive.i32.html`
core/primitive.i32.html:1262: broken link - `primitive.reference.html`
core/primitive.i32.html:1263: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1264: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i128.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i16.html`
core/primitive.i32.html:1268: broken link - `primitive.usize.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1268: broken link - `primitive.i32.html`
core/primitive.i32.html:1269: broken link - `primitive.i32.html`
```

</details>

gurry commented on src/librustdoc/html/format.rs:
```diff
@@ -879,11 +879,16 @@ fn primitive_link_fragment(
+                    let cname_sym = ExternalCrate { crate_num: def_id.krate }.name(cx.tcx());
+                    format!("{cname_sym}/")
+                } else {
+                    "../".repeat(len)
```
Uh oh

gurry committed:
Rustdoc: include crate name in links for local primitives

It makes the link easier to use in cases in which
the path of the page where it will be embedded is not
known beforehand such as when we generate impls
dynamically from `register_type_impls` method in
`main.js`

Earlier for local primitives we would generate a path
that was relative to the current page depth passed in `cx.current`
. e.g if the current page was `std::simd::prelude::Simd` the
generated path would be `../../primitive.<prim>.html`  After this
change the path will first take you to the the wesite root and add
the crate name. e.g. for `std::simd::prelude::Simd` the path now
will be `../../../std/primitive.<prim>.html`

bors commented:
:pushpin: Commit e0bfa5c2940eb926c3b7d8d85f6f351a767d6664 has been approved by `notriddle`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=notriddle e0bfa5c2940eb926c3b7d8d85f6f351a767d6664 -->
<!-- homu: {"type":"Approved","sha":"e0bfa5c2940eb926c3b7d8d85f6f351a767d6664","approver":"notriddle","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:evergreen_tree: The tree is currently [closed](https://github.com/rust-lang/rust/pull/121523#issuecomment-1962124798) for pull requests below priority 50. This pull request will be tested once the tree is reopened.
